src/views/ProductView.vue

```diff
@@ -578,4 +622,247 @@ function goToWishlists() {
+    if (inventoryStatusOfProductItems.value !== null) {
+        return inventoryStatusOfProductItems.value.filter(
+            (productItem) =>
+                productItem.inventoryStatus === ProductItemStatus.InFulfillment
+        ).length
+    }
+
+    return undefined
+})
+
+/**
+ * Computed property for the number of shipped product items.
+ * @returns The number of shipped product items or undefined if no data is available.
+ */
+const numberOfShippedProductItems = computed(() => {
+    if (inventoryStatusOfProductItems.value !== null) {
+        return inventoryStatusOfProductItems.value.filter(
+            (productItem) =>
+                productItem.inventoryStatus === ProductItemStatus.Shipped
+        ).length
+    }
+
+    return undefined
+})
+
+/**
+ * Computed property for the number of delivered product items.
+ * @returns The number of delivered product items or undefined if no data is available.
+ */
+const numberOfDeliveredProductItems = computed(() => {
+    if (inventoryStatusOfProductItems.value !== null) {
+        return inventoryStatusOfProductItems.value.filter(
+            (productItem) =>
+                productItem.inventoryStatus === ProductItemStatus.Delivered
+        ).length
+    }
+
+    return undefined
+})
+
+/**
+ * Computed property for the number of returned product items.
+ * @returns The number of returned product items or undefined if no data is available.
+ */
+const numberOfReturnedProductItems = computed(() => {
+    if (inventoryStatusOfProductItems.value !== null) {
+        return inventoryStatusOfProductItems.value.filter(
+            (productItem) =>
+                productItem.inventoryStatus === ProductItemStatus.Returned
+        ).length
+    }
+
+    return undefined
+})
+
+/**
+ * Computed property for the number of lost product items.
+ * @returns The number of lost product items or undefined if no data is available.
+ */
+const numberOfLostProductItems = computed(() => {
+    if (inventoryStatusOfProductItems.value !== null) {
+        return inventoryStatusOfProductItems.value.filter(
+            (productItem) =>
+                productItem.inventoryStatus === ProductItemStatus.Lost
+        ).length
+    }
+
+    return undefined
+})
+
+/**
+ * The maximum number of product items a buyer can order.
+ *
+ * This computed ref calculates the maximum number of product items a buyer can order based on the
+ * number of items in stock and the number of reserved items.
+ * The available quantity is the number of items in stock minus the number of reserved items.
+ * If the available quantity is less than 10, it returns that value; otherwise,
+ * it returns the default maximum value of 10.
+ */
+const maximumNumberOfProductItemsABuyerCanOrder = computed(() => {
+    const defaultValue = 10
+
+    if (
+        numberOfProductItemsInStock.value !== undefined &&
+        numberOfReservedProductItems.value !== undefined
+    ) {
+        const numberOfAvailableProductItems =
```

Comment:
a product item has only one state
so there cannot be any overlap between in storage and reserved ones
thus I think the correct calculation would be just taking the  `numberOfProductItemsInStock` ones

Reply:
-> 821d4dd7f52015e8a6f78424b916ca72ccf895a2